src/client/connect/http.rs

```diff
@@ -338,8 +338,9 @@ where
 
         let sock = c.connect().await?;
 
-        sock.set_nodelay(config.nodelay)
-            .map_err(ConnectError::m("tcp set_nodelay error"))?;
+        if sock.set_nodelay(config.nodelay).is_err() {
```

Comment:
Probably worth logging the actual error too, by making this `if let Err(err) = ...`, and then having `warn!("tcp set_nodelay error: {}", err);`.